src/main/java/bio/terra/service/dataset/flight/datadelete/DropExternalTablesStep.java

```diff
@@ -0,0 +1,62 @@
+package bio.terra.service.dataset.flight.datadelete;
+
+import bio.terra.model.DataDeletionRequest;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.job.JobMapKeys;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.UUID;
+
+
+public class DropExternalTablesStep implements Step {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final DatasetService datasetService;
+
+    private static Logger logger = LoggerFactory.getLogger(DropExternalTablesStep.class);
+
+    public DropExternalTablesStep(BigQueryPdao bigQueryPdao, DatasetService datasetService) {
+        this.bigQueryPdao = bigQueryPdao;
+        this.datasetService = datasetService;
+    }
+
+    private String suffix(FlightContext context) {
+        return context.getFlightId().replace('-', '_');
+    }
+
+    private DataDeletionRequest request(FlightContext context) {
+        return context.getInputParameters()
+            .get(JobMapKeys.REQUEST.getKeyName(), DataDeletionRequest.class);
+    }
+
+    private Dataset dataset(FlightContext context) {
```

Comment:
I think this method is doing the same thing as the DataDeletionUtils method. Maybe you meant to replace it?

Reply:
Ah I thought I factored this out. Thanks, good catch!